ssh-key/src/sshsig.rs

```diff
@@ -113,18 +131,45 @@ impl SshSig {
     ///
     /// Once a [`Signature`] has been computed over the returned byte vector,
     /// [`SshSig::new`] can be used to construct the final signature.
-    pub fn signed_data(namespace: &str, hash_alg: HashAlg, msg: &[u8]) -> Result<Vec<u8>> {
+    pub fn signed_data(
+        namespace: &str,
+        hash_alg: HashAlg,
+        msg: &[u8],
+        application: Option<&[u8]>,
+        flags: Option<u8>,
+        counter: Option<u32>,
+    ) -> Result<Vec<u8>> {
```

Comment:
This is a public API, so this is a breaking change.

Perhaps you could introduce a separate (possibly private) function that `signed_data` is composed in terms of?